app/assets/javascripts/leaflet.map.js

```diff
@@ -76,12 +76,31 @@ L.OSM.Map = L.Map.extend({
 
     /* see also timeslider.js and viewreset/baselayerchange handlers */
 
+    // Add multi-lingual awareness
+    const language = new MapboxLanguage({
+      defaultLanguage: 'mul'
+    });
+    console.info(`leaflet.map: preferred_languages: ${OSM.preferred_languages}`);
+    console.info(`  browser_language: ${navigator.language}`);
+    let selectedLanguage, plFound = false;
+    for (let i=0; i<OSM.preferred_languages.length; i++) {
+      // Strip variant, e.g., '_US', as none are to be found in supportedLanguages. Loop is broken so deduping preferred_languages unnecessary.
+      OSM.preferred_languages[i] = OSM.preferred_languages[i].replace(/-.*$/,'')
+      if (language.supportedLanguages.includes(OSM.preferred_languages[i])) {
+        selectedLanguage = OSM.preferred_languages[i];
+        plFound = true;
+        break
+      }
+      if (!plFound) selectedLanguage = navigator.language.replace(/-.*$/,'')
```

Comment:
Should this line go outside of the for loop? `plFound` is guaranteed to be false by the time we get here, but it looks like this is intended to be a fallback in case we get through the entire loop without setting `selectedLanguage`.

Reply:
The entire validation block was removed as we accept any language string specified by a user.